go/multipooler/manager/rpc_manager.go

```diff
@@ -909,6 +909,83 @@ func (pm *MultiPoolerManager) Demote(ctx context.Context, consensusTerm int64, d
 	}, nil
 }
 
+// demoteLocked performs the core demotion logic.
+// REQUIRES: action lock must already be held by the caller.
+// This is used by BeginTerm to demote inline without re-acquiring the lock.
+func (pm *MultiPoolerManager) demoteLocked(ctx context.Context, consensusTerm int64, drainTimeout time.Duration) (string, error) {
```

Comment:
I think the Demote rpc, should also use htis method now. So we don't have duplication.